### PR TITLE
Lido app/marcus/xlookup

### PR DESCRIPTION
**Pull request by marhong2**
XLOOKUP is done, I also added WC match to the dependency list @tyleradams

### CODE REVIEW

File: formulas/helpers.js [hunk 1]
```diff
@@ -369,6 +370,76 @@ class FormulaHelpers {
...
+        return direction * longer_string.charCodeAt(min_Index + 1);
+      }
+      return 0;
+    }
+    //The special cases with the unique regex values
+    if (!match_mode) {
+      if (search_mode) {
+        throw FormulaError.VALUE;
+      }
+      //EXCEL uses ~ instead of \ for their escape character, so we replace all instances of ~ with \ so that we can use a regex comparison with wcmatch
```
Comment (tyleradams) [inline formulas/helpers.js:436]: Oh slick

**Review by tyleradams — CHANGES_REQUESTED** [review]
Looking good!

**Comment by marhong2** [comment]
Updated @tyleradams

**Comment by tyleradams** [comment]
Looks good to me!

**Comment by mary-tung** [comment]
Is there any reason why XLOOKUP() wouldn't be showing up in our app? Looks like all the other recently added formulas are working fine

image

**Comment by tyleradams** [comment]
https://linear.app/trylido/issue/LIDO-3340/document-these-methods XLOOKUP
is on the list

**Pull request merged**